File: app/src/test/java/com/babestudios/companyinfouk/ui/insolvency/InsolvencyPresenterTest.java
package com.babestudios.companyinfouk.ui.insolvency;

import com.babestudios.companyinfouk.data.DataManager;
import com.babestudios.companyinfouk.data.model.insolvency.Insolvency;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

import rx.Observable;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class InsolvencyPresenterTest {

	private InsolvencyPresenter insolvencyPresenter;

	@Before
	public void setUp() {
		insolvencyPresenter = new InsolvencyPresenter(mock(DataManager.class));
		insolvencyPresenter.create();
		InsolvencyActivityView view = mock(InsolvencyActivityView.class);
		when(view.getCompanyNumber()).thenReturn("0");
		insolvencyPresenter.attachView(view);
		when(insolvencyPresenter.dataManager.getInsolvency("0")).thenReturn(Observable.just(new Insolvency()));
	}

	@Test
	public void test_When_GetInsolvency_Then_DataManagerGetInsolvencyIsCalled() {
		insolvencyPresenter.getInsolvency();
		verify(insolvencyPresenter.dataManager).getInsolvency("0");
	}
}

File: app/src/test/java/com/babestudios/companyinfouk/ui/insolvency/InsolvencyPresenterTest.kt
package com.babestudios.companyinfouk.ui.insolvency

import com.babestudios.companyinfouk.CompaniesHouseApplication
import com.babestudios.companyinfouk.DaggerTestApplicationComponent
import com.babestudios.companyinfouk.TestApplicationModule
import com.babestudios.companyinfouk.data.model.insolvency.Insolvency
import com.nhaarman.mockitokotlin2.whenever
import io.reactivex.CompletableSource
import io.reactivex.Single
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito.*
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class InsolvencyPresenterTest {

	private lateinit var insolvencyPresenter: InsolvencyPresenter

	@Before
	fun setUp() {
		val testApplicationComponent = DaggerTestApplicationComponent.builder()
				.testApplicationModule(TestApplicationModule(CompaniesHouseApplication()))
				.build()
		insolvencyPresenter = testApplicationComponent.insolvencyPresenter()
		whenever(insolvencyPresenter.companiesRepository.getInsolvency("0")).thenReturn(Single.just(Insolvency()))
		val insolvencyViewModel = InsolvencyViewModel()
		insolvencyViewModel.state.value.companyNumber = "0"
		insolvencyPresenter.setViewModel(insolvencyViewModel, CompletableSource { })
	}

	@Test
	fun test_When_GetInsolvency_Then_DataManagerGetInsolvencyIsCalled() {
		insolvencyPresenter.fetchInsolvencies("0")
		verify(insolvencyPresenter.companiesRepository, times(2)).getInsolvency("0")
	}
}